src/product_config_utils.rs

```diff
@@ -418,31 +413,17 @@ where
 fn parse_cli_properties<T>(
     resource: &<T as Configuration>::Configurable,
     role_name: &str,
-    config: Option<&CommonConfiguration<T>>,
+    config: &CommonConfiguration<T>,
 ) -> BTreeMap<String, Option<String>>
 where
     T: Configuration,
 {
-    let mut final_properties = BTreeMap::new();
-
     // Properties from the role have the lowest priority, so they are computed and added first...
-    if let Some(CommonConfiguration {
-        config: Some(ref config),
-        ..
-    }) = config
-    {
-        final_properties = config.compute_cli(resource, role_name).unwrap();
-    }
+    let mut final_properties = config.config.compute_cli(resource, role_name).unwrap();
```

Comment:
It was in before, but i think we should not just unwrap here (and in parse_env_properties, parse_file_properties).
Though we have it under control via framework and operators, come to think of it maybe we want to throw an error if some config properties are wrong, bad combination etc. So i think we should handle this here or at least use unwrap_or_default()?